### PR TITLE
Support update k8s version

### PR DESCRIPTION
**Pull request by smallteeths**
https://github.com/cnrancher/pandaria/issues/2565

### CODE REVIEW

File: controller/ack-cluster-config-handler.go [hunk 1]
```diff
@@ -227,11 +247,55 @@ func (h *Handler) checkAndUpdate(config *ackv1.ACKClusterConfig) (*ackv1.ACKClus
 	if err != nil {
 		return config, err
 	}
-
 	clusterState := utils.GetMapString("state", *cluster)
+	if err != nil {
+		return config, err
+	}
```
Comment (GGGitBoy) [inline controller/ack-cluster-config-handler.go]: 这部分处理看似没有必要

File: controller/ack-cluster-config-handler.go [hunk 2]
```diff
@@ -227,11 +247,55 @@ func (h *Handler) checkAndUpdate(config *ackv1.ACKClusterConfig) (*ackv1.ACKClus
...
+
+	// ACK k8s version update
+	if !clusterIsUpgrading &&
+		!config.Spec.PauseClusterUpgrade &&
+		!config.Spec.ClusterIsUpgrading &&
+		(config.Status.Phase == ackConfigActivePhase || strings.Contains(config.Status.FailureMessage, ack.UpdateK8SVersionApiError)) {
+		if config.Spec.KubernetesVersion != utils.GetMapString("current_version", *cluster) {
+			config.Status.Phase = ackConfigUpdatingPhase
+			err = ack.UpgradeCluster(client, &config.Spec)
+			if err != nil {
```
Comment (GGGitBoy) [inline controller/ack-cluster-config-handler.go]: if err = ack.UpgradeCluster(client, &config.Spec); err != nil {

File: controller/ack-cluster-config-handler.go (hunk 1 above)
Comment (smallteeths) [reply controller/ack-cluster-config-handler.go]: done

File: controller/ack-cluster-config-handler.go (hunk 2 above)
Comment (smallteeths) [reply controller/ack-cluster-config-handler.go]: done

**Comment by GGGitBoy** [comment]
LGTM

**Pull request merged**